### PR TITLE
Added a TFDS and Croissant notebook

### PR DESCRIPTION
**Pull request by ccl-core**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -75,11 +75,14 @@ jobs:
         ipython kernel install --user --name croissant-notebook
         for notebook in recipes/*ipynb
           do
-            jupyter nbconvert \
-              --ExecutePreprocessor.timeout=600 \
-              --ExecutePreprocessor.kernel_name=croissant-notebook \
-              --to notebook \
-              --execute $notebook
+            if [[ "$notebook" != "recipes/tfds_croissant_builder.ipynb" ]]
```
Comment (marcenacp) [inline .github/workflows/ci.yml:78]: Why not test it?

Comment (ccl-core) [reply .github/workflows/ci.yml:78]: The PyTorch part doesn't work yet

File: python/mlcroissant/recipes/tfds_croissant_builder.ipynb [hunk 2]
```diff
@@ -0,0 +1,1540 @@
...
+        },
+        "id": "Nxg8vH-yvlD8",
+        "outputId": "4d7be517-b98d-4fd8-db68-79bef2c5e881"
+      },
+      "outputs": [
+        {
+          "name": "stderr",
+          "output_type": "stream",
+          "text": [
+            "WARNING:absl:Found the following 1 warning(s) during the validation:\n",
```
Comment (marcenacp) [inline python/mlcroissant/recipes/tfds_croissant_builder.ipynb:147]: Can you clean the output?

File: python/mlcroissant/recipes/tfds_croissant_builder.ipynb [hunk 3]
```diff
@@ -0,0 +1,1540 @@
...
+          "text": [
+            "Downloading builder script: 100% 4.83k/4.83k [00:00<00:00, 17.0MB/s]\n",
+            "Downloading metadata: 100% 3.13k/3.13k [00:00<00:00, 10.9MB/s]\n",
+            "Downloading readme: 100% 8.85k/8.85k [00:00<00:00, 23.6MB/s]\n",
+            "I1127 13:47:02.165981 140028632248320 from_huggingface_to_croissant.py:189] Done. Wrote Croissant JSON-LD to /tmp/croissant_1701092817.7705376.json\n"
+          ]
+        }
+      ],
+      "source": [
+        "!mlcroissant from_huggingface_to_croissant --dataset fashion_mnist"
```
Comment (marcenacp) [inline python/mlcroissant/recipes/tfds_croissant_builder.ipynb:109]: This script will be removed soon. Can you directly declare the Croissant file?

File: python/mlcroissant/recipes/tfds_croissant_builder.ipynb [hunk 4]
```diff
@@ -0,0 +1,1540 @@
...
+            "WARNING:absl:Found the following 1 warning(s) during the validation:\n",
+            "  -  [dataset(fashion_mnist)] Property \"https://schema.org/license\" is recommended, but does not exist.\n",
+            "WARNING:absl:Using custom data configuration default\n"
+          ]
+        }
+      ],
+      "source": [
+        "fashion_croissant_path = \"/tmp/croissant_1701092817.7705376.json\"\n",
+        "\n",
+        "fashion_mnist_builder = CroissantBuilder(\n",
```
Comment (marcenacp) [inline python/mlcroissant/recipes/tfds_croissant_builder.ipynb:156]: WDYT of:

`fashion_mnist_builder`>`builder`
`fashion_croissant_path`>`file`
etc

To have shorter/easier to read names?

File: python/mlcroissant/recipes/tfds_croissant_builder.ipynb [hunk 5]
```diff
@@ -0,0 +1,1540 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "n92WWWKk6o1I"
+      },
+      "source": [
+        "# TensorFlow Datasets for Croissant 🥐\n",
```
Comment (marcenacp) [inline python/mlcroissant/recipes/tfds_croissant_builder.ipynb:9]: `Train a model with Croissant and TFDS`?

File: python/mlcroissant/recipes/tfds_croissant_builder.ipynb (hunk 2 above)
Comment (ccl-core) [reply python/mlcroissant/recipes/tfds_croissant_builder.ipynb:147]: Done.

File: python/mlcroissant/recipes/tfds_croissant_builder.ipynb (hunk 3 above)
Comment (ccl-core) [reply python/mlcroissant/recipes/tfds_croissant_builder.ipynb:109]: Sure. I submitted the file under `datasets/`

File: python/mlcroissant/recipes/tfds_croissant_builder.ipynb (hunk 5 above)
Comment (ccl-core) [reply python/mlcroissant/recipes/tfds_croissant_builder.ipynb:9]: Done.

**Comment by ccl-core** [comment]
@marcenacp a first version of the notebook. I will add a further section with a working example of training a model in a subsequent PR.

File: python/mlcroissant/recipes/tfds_croissant_builder.ipynb [hunk 6]
```diff
@@ -0,0 +1,1354 @@
...
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "!git clone https://github.com/mlcommons/croissant.git\n",
+        "!cd croissant/python/mlcroissant\n",
```
Comment (marcenacp) [inline python/mlcroissant/recipes/tfds_croissant_builder.ipynb:49]: Apparently we have to write `%cd` here

**Review by marcenacp — APPROVED** [review]
Thanks!

File: python/mlcroissant/recipes/tfds_croissant_builder.ipynb (hunk 6 above)
Comment (ccl-core) [reply python/mlcroissant/recipes/tfds_croissant_builder.ipynb]: That's true, done!

**Pull request merged**